### PR TITLE
fill the complete address because of no userguide directory

### PR DESCRIPTION
**Pull request by lixiaobing1**
fill the complete address because of no userguide directory

### CODE REVIEW

File: docs/reference/api/hub_registry_spec.md [hunk 1]
```diff
@@ -20,4 +20,4 @@ This API is deprecated as of 1.7. To view the old version, see the [go
 here](https://docs.docker.com/v1.7/docker/reference/api/hub_registry_spec/) in
 the 1.7 documentation. If you want an overview of the current features in
 Docker Hub or other image management features see the [image management
-overview](../../userguide/eng-image/image_management.md) in the current documentation set.
+overview](https://docs.docker.com/engine/userguide/eng-image/image_management.md) in the current documentation set.
```
Comment (thaJeztah) [inline docs/reference/api/hub_registry_spec.md]: should be without `.md`; `https://docs.docker.com/engine/userguide/eng-image/image_management/`

File: docs/reference/builder.md [hunk 2]
```diff
@@ -23,7 +23,7 @@ instructions in succession.
 
 This page describes the commands you can use in a `Dockerfile`. When you are
 done reading this page, refer to the [`Dockerfile` Best
-Practices](../userguide/eng-image/dockerfile_best-practices.md) for a tip-oriented guide.
+Practices](https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices.md) for a tip-oriented guide.
```
Comment (thaJeztah) [inline docs/reference/builder.md]: same here (no `.md`)

File: docs/reference/builder.md [hunk 3]
```diff
@@ -92,7 +92,7 @@ instructions.
 Whenever possible, Docker will re-use the intermediate images (cache),
 to accelerate the `docker build` process significantly. This is indicated by
 the `Using cache` message in the console output.
-(For more information, see the [Build cache section](../userguide/eng-image/dockerfile_best-practices.md#build-cache)) in the
+(For more information, see the [Build cache section](https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices.md#build-cache)) in the
```
Comment (thaJeztah) [inline docs/reference/builder.md]: should be `https://docker.github.io/engine/userguide/eng-image/dockerfile_best-practices/#/build-cache`

File: docs/reference/builder.md [hunk 4]
```diff
@@ -574,7 +574,7 @@ cache for `RUN` instructions can be invalidated by using the `--no-cache`
 flag, for example `docker build --no-cache`.
 
 See the [`Dockerfile` Best Practices
-guide](../userguide/eng-image/dockerfile_best-practices.md#build-cache) for more information.
+guide](https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices.md#build-cache) for more information.
```
Comment (thaJeztah) [inline docs/reference/builder.md]: same here; `https://docker.github.io/engine/userguide/eng-image/dockerfile_best-practices/#/build-cache`

File: docs/reference/builder.md [hunk 5]
```diff
@@ -830,7 +830,7 @@ of whether or not the file has changed and the cache should be updated.
 > following instructions from the Dockerfile if the contents of `<src>` have
 > changed. This includes invalidating the cache for `RUN` instructions.
 > See the [`Dockerfile` Best Practices
-guide](../userguide/eng-image/dockerfile_best-practices.md#build-cache) for more information.
+guide](https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices.md#build-cache) for more information.
```
Comment (thaJeztah) [inline docs/reference/builder.md]: same here `https://docker.github.io/engine/userguide/eng-image/dockerfile_best-practices/#/build-cache`

**Review by thaJeztah — CHANGES_REQUESTED** [review]
left some comments

**Comment by lixiaobing10051267** [comment]
ok, has been modified:)

**Review by thaJeztah — APPROVED** [review]
LGTM

**Review by mdlinville — APPROVED** [review]
LGTM, thank you!

**Pull request merged**